### ISSUE TITLE
Lock on our strategy for templating for kubernets manfiests

### ISSUE THREAD

**Issue by ellismg** [Azure/azure-dev#1649]
Our deployment strategy for Kubernetes is to allow the user to provide a series of k8s manifests which we `kubectl apply` in sorted order.  In order to pass information from `azd` into these manifests, we allow replacement using `envsubst` style replacements.  This allows the user to do something like:

```
    spec:
      containers:
        - name: todo-web
          image: ${SERVICE_WEB_IMAGE_NAME}
```

One concern is that this file might contain other things that look like shell replacements and (e.g. inline scripts) and this strategy may end up wrecking them.  Some discussion was raised on the PR: https://github.com/Azure/azure-dev/pull/1629#discussion_r1122296885.

Long term we will add additional deployment methods which may have more first-class support for parameterizing files (e.g. helm or others) but we should think about what we want our strategy to be here.  We may want to have a way to say: "here's a manifest, please do not process it, just send it as is".

**Comment by wbreza** [comment]
Linking related PR #2455

**Comment by wbreza** [comment]
Resolved by #2455
